substratevm/mx.substratevm/testhello.py

```diff
@@ -124,6 +124,10 @@ def execute(command):
 
 def test():
 
+    # define some useful constants
+    main_start = 215
```

Comment:
To avoid all those line-number change related changes we should use anchors.

Define a python function like
```python
def lineNumberOf(filename, regex):
    ....
    return line_number_where_given_regex_matches_or_error_if_not_found_or_found_more_than_once
```

Reply:
With something like that we could prevent the need for changes just because some part of `Hello.java` has new lines somewhere in the beginning.

Reply:
We already do something like this for checking listings and backtraces, by not specifying the line number and expecting a pattern that matches the line we want, e.g.: https://github.com/oracle/graal/blob/3e7732200f48d72a434e69b0d89ffa997e7d23fb/substratevm/mx.substratevm/testhello.py#L214

As a result this suggestion would only make sense for setting breakpoints at specific lines like we do in: https://github.com/oracle/graal/blob/3e7732200f48d72a434e69b0d89ffa997e7d23fb/substratevm/mx.substratevm/testhello.py#L605 

https://github.com/oracle/graal/blob/3e7732200f48d72a434e69b0d89ffa997e7d23fb/substratevm/mx.substratevm/testhello.py#L630

and

https://github.com/oracle/graal/blob/3e7732200f48d72a434e69b0d89ffa997e7d23fb/substratevm/mx.substratevm/testhello.py#L661 

The reason we explicitly set the constants is to be more verbatim on what we expect and be able to catch glitches that might otherwise get away (e.g. line numbers being wrong by one).

Reply:
@olpaw WDYT about ^^

Reply:
> The reason we explicitly set the constants is to be more verbatim on what we expect and be able to catch glitches that might otherwise get away (e.g. line numbers being wrong by one).

Alright. I leave it up to you if you want to keep explicit line numbers here.